src/python_testing/mdns_discovery/mdns_service_info.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class MdnsServiceInfo:
+    # The unique name of the mDNS service.
+    service_name: str
+
+    # The human-readable name of the service.
+    name: str
+
+    # The type of the service, typically indicating the service protocol and domain.
+    type: str
+
+    # The domain name of the machine hosting the service.
+    server: str
+
+    # The network port on which the service is available.
+    port: int
+
+    # A list of IP addresses associated with the service.
+    addresses: list
+
+    # A dictionary of key-value pairs representing the service's metadata.
+    txt_record: dict
+
+    # The priority of the service, used in service selection when multiple instances are available.
+    priority: int
+
+    # The network interface index on which the service is advertised.
+    interface_index: int
+
+    # The relative weight for records with the same priority, used in load balancing.
+    weight: int
+
+    # The time-to-live value for the host name in the DNS record.
+    host_ttl: int
+
+    # The time-to-live value for other records associated with the service.
+    other_ttl: int
```

Comment:
Is this really a list/dict? Otherwise, which record's TTL lands here?

Reply:
This is for each record, maybe it was a type in the description or return type